packages/storage-plus/src/snapshot/map.rs

```diff
@@ -262,6 +262,33 @@ where
     }
 }
 
+#[cfg(feature = "iterator")]
+impl<'a, K, T> SnapshotMap<'a, K, T>
+where
+    T: Serialize + DeserializeOwned + Clone,
+    K: PrimaryKey<'a> + Prefixer<'a> + KeyDeserialize,
+{
+    pub fn changelog_range<'c>(
+        &self,
+        store: &'c dyn Storage,
+        k: &K,
+        min: Option<Bound>,
+        max: Option<Bound>,
+        order: cosmwasm_std::Order,
+    ) -> Box<dyn Iterator<Item = StdResult<(u64, ChangeSet<T>)>> + 'c>
+    where
+        T: 'c,
+        'a: 'c,
+        K: 'c,
+        K::Output: 'static,
+    {
+        self.snapshots
+            .changelog
+            .prefix(k.clone())
+            .range(store, min, max, order)
```

Comment:
You are combining / hard-coding `prefix` and `range` here.

As mentioned above, instead of doing that it would be better to follow the `range` contract for `changelog_range` as well. And also add a `changelog_prefix` method.

Now, I agree with your comment that making `snapshot` and `changelog` public will basically provide the same, without requiring these wrapper / accessor methods.

Perhaps simply writing a `changelog()` method is the best approach here. It will return the underlying changelog map, and then you can prefix / iterate over it as you see fit.

@ethanfrey , what do you think?

Reply:
> Perhaps simply writing a `changelog()` method is the best approach here. It will return the underlying changelog map, and then you can prefix / iterate over it as you see fit.

Of course, this would allow writing to the changelog directly, which is wrong. But, you can always do that anyway, by playing with the namespaces. So, I don't see this is a real stopper here.

Reply:
Best thing would be to return a `MapView` struct, that doesn't have the `save` / `update` / `remove` methods, but I think that's out of scope here.

Reply:
See #622 for an implementation of this approach.

Reply:
amazing! Thank you so much for the help! 

I think that is a great approach.